src/crypto_helper/input/bcrypt.rs

```diff
@@ -0,0 +1,133 @@
+        match event.target_unchecked_into::<HtmlInputElement>().value().parse::<u32>() {
+            Ok(rounds) => {
+                if let BcryptAction::Hash(bcrypt_hash_action) = bcrypt_action.clone() {
+                    input_setter.emit(BI {
+                        data: data.clone(),
+                        action: BcryptAction::Hash(BcryptHashAction {
+                            rounds,
+                            salt: bcrypt_hash_action.salt,
+                        })
+                    })
+                }
+            },
+            Err(_) => (),
+        };
+    });
+
+    let is_valid = use_state(|| true);
+    let set_is_valid = is_valid.setter();
+
+    let input_setter = input_props.bcrypt_input_setter.clone();
+    let bcrypt_action = input_props.input.action.clone();
+    let data = input_props.input.data.clone();
+
+    let on_salt_input = Callback::from(move |salt: Vec<u8>| {
+        if let BcryptAction::Hash(hash_action) = bcrypt_action.clone() {
+            match salt.len() {
+                16 | 0 => {
+                    set_is_valid.set(true);
+                },
+                _ => {
+                    set_is_valid.set(false);
+                }
+            };
+            input_setter.emit(BI {
+                data: data.clone(),
+                action: BcryptAction::Hash(BcryptHashAction {
+                    salt,
+                    rounds: hash_action.rounds,
+                })
+            })
+        }
+    });
+
+    let bcrypt_action = input_props.input.action.clone();
+    let input_setter = input_props.bcrypt_input_setter.clone();
+    let byte_setter = Callback::from(move |data| {
+        input_setter.emit(BI {
+            action: bcrypt_action.clone(),
+            data,
+        });
+    });
+
+    let input_setter = input_props.bcrypt_input_setter.clone();
+    let data = input_props.input.data.clone();
+    let input = input_props.input.clone();
+    let on_switch = Callback::from(move |mode: bool| {
+        let BI { action: _, data } = input.clone();
+        input_setter.emit(BI {
+            data,
+            action: mode.into(),
+        });
+    });
+
+    let data = input_props.input.data.clone();
+    let input_setter = input_props.bcrypt_input_setter.clone();
+    let bcrypt_hash_action = input_props.input.action.clone();
+    let on_hashed_input = Callback::from(move |hashed: Vec<u8>| {
+        if let BcryptAction::Verify(_) = bcrypt_hash_action.clone() {
+            input_setter.emit(BI {
+                data: data.clone(),
+                action: BcryptAction::Verify(std::str::from_utf8(&hashed).unwrap_or("").to_string()),
+            })
+        }
+    });
+
+    let action = input_props.input.action.clone();
+    let data = input_props.input.data.clone();
+    html! {
+        <div class={classes!("formats-container", "vertical")}>
+
+            {build_byte_input(data, byte_setter, None, Some("bcrypt".into()))}
```

Comment:
usually, users enter the plain UTF-8 password to hash. so let's set the `Ascii` format for it by default

```suggestion
            {build_byte_input(data, byte_setter, Some(BytesFormat::Ascii), Some("bcrypt".into()))}
```